pyuvdata/uvdata/miriad.py

```diff
@@ -1014,48 +1051,86 @@ def read_miriad(
                 cnt = np.ones(d.shape, dtype=np.float64)
             ra = uv["ra"]
             dec = uv["dec"]
-            # NOTE: Using our lst calculator, which uses astropy,
-            # instead of _miriad values which come from pyephem.
-            # The differences are of order 5 seconds.
-            # To use the values from the file you'd want: lst = uv['lst']
+            if record_epoch:
+                epoch = uv["epoch"]
+            if record_app:
+                app_ra = uv["obsra"]
+                app_dec = uv["obsdec"]
+            if record_pa:
+                frame_pa = uv["obspa"]
+            # LST is pulled from the file here, atlhough as some PAPER/early HERA
+            # data calculated LST from pyephem (which is inconsistent w/ astropy
+            # to the order of ~5 seconds), these values can be recalculated by
+            # setting `calc_lst=True` when calling read_miriad.
+            lst = uv["lst"]
             inttime = uv["inttime"]
-            source = uv["source"]
-            if source != _source:
-                if skip_extra_sources:
-                    if warn_extra_sources:
-                        warnings.warn(
-                            "File containts more than one source, only using data "
-                            "where source = %s" % _source
-                        )
-                        warn_extra_sources = False
-                    continue
-                raise NotImplementedError(
-                    "This appears to be a multi source file, which is not supported. "
-                    "You can bypass this error bu using skip_extra_sources=True to "
-                    "read in only the data belonging to the first source in the data "
-                    "set (and skip data from other sources)."
-                )
-            else:
-                _source = source
+            try:
+                sou_id = sou_dict[uv["source"]]
+            except KeyError:
+                sou_dict[uv["source"]] = Nphase
+                sou_id = Nphase
+                Nphase += 1
+
+            # TODO: The code below serves literally no purpose at the moment, as the
+            # values in check_variables are not used past this point. Commenting it out
+            # until it can be refactored at some point.
 
             # check extra variables for changes compared with initial value
-            for extra_variable in list(check_variables.keys()):
-                if type(check_variables[extra_variable]) == str:
-                    if uv[extra_variable] != check_variables[extra_variable]:
-                        check_variables.pop(extra_variable)
-                else:
-                    if not np.allclose(
-                        uv[extra_variable], check_variables[extra_variable]
-                    ):
-                        check_variables.pop(extra_variable)
+            # for extra_variable in list(check_variables.keys()):
```

Comment:
So I commented it out because the the output here (`check_variables`) is literally never being looked at or used past this point, and when these "extra variables" exist up in MIRIAD files (e.g., in the CARMA test file), this loop is by far the slowest part of the file read. I _think_ it was originally intended to check for record-varying vs static values in MIRIAD files, although because the output never gets used, `UVData` objects just capture the first entry.  I could just delete the defunct block of code and file an issue, if that'd be preferred.

Reply:
Filed an issue, and have removed the commented out code in 72897c0c.